server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/rewrite/selector/RewriteCompactionFileSelector.java

```diff
@@ -229,6 +235,29 @@ void selectFiles() throws IOException {
     }
   }
 
+  /**
+   * If the unseq file does not overlap with any seq files, then select the latest sealed seq file
+   * for it to compact with. Notice: If the data is deleted and then start an inner space
+   * compaction, it may cause unseqStartTime > seqEndTime or partial devices are in the unseq files
+   * but not in seq files, which will cause the unseq files to not overlap with any seq file
+   */
+  private boolean hasFoundSeqFiles(TsFileResource unseqFile) {
+    if (!tmpSelectedSeqFiles.isEmpty()) {
+      return true;
+    }
+    // the current unseq file does not overlap with any seq files
+    logger.info("Unseq file {} does not overlap with seq files.", unseqFile);
```

Comment:
Suggest to make line 250-257 as a separate method such as `trySelectLatestSealedSeqFile()`. And then change the judgement above into `if (!hasFoundSeqFiles(unseqFile) && !trySelectLatestSealedSeqFile())`